packages/medusa/src/api/routes/admin/notifications/list-notifications.ts

```diff
@@ -0,0 +1,182 @@
+ *     name: expand
+ *     schema:
+ *       type: string
+ *     description: The fields to populate
+ *   - in: query
+ *     name: event_name
+ *     schema:
+ *       type: string
+ *   - in: query
+ *     name: resource_type
+ *     schema:
+ *       type: string
+ *   - in: query
+ *     name: resource_id
+ *     schema:
+ *       type: string
+ *   - in: query
+ *     name: to
+ *     schema:
+ *       type: string
+ *   - in: query
+ *     name: include_resends
+ *     schema:
+ *       type: boolean
+ *     description: Whether the result set should include resends or not
+ *
+ * tags:
+ *   - Notification
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           properties:
+ *             notifications:
+ *               type: array
+ *               items:
+ *                 $ref: "#/components/schemas/notification"
+ */
+export default async (req, res) => {
+  const notificationService: NotificationService = req.scope.resolve(
+    "notificationService"
+  )
+  const validatedQuery = await validator(AdminGetNotificationsQuery, req.query)
+
+  const limit = validatedQuery.limit ? parseInt(validatedQuery.limit) : 50
+  const offset = validatedQuery.offset ? parseInt(validatedQuery.offset) : 0
+
+  const selector: any = {}
+
+  let includeFields: string[] = []
+  if (validatedQuery.fields) {
+    includeFields = validatedQuery.fields.split(",")
+  }
+
+  let expandFields: string[] = []
+  if (validatedQuery.expand) {
+    expandFields = validatedQuery.expand.split(",")
+  }
+
+  if (validatedQuery.event_name) {
+    const values = validatedQuery.event_name.split(",")
+    selector.event_name = values.length > 1 ? values : values[0]
+  }
+
+  if (validatedQuery.resource_type) {
+    const values = validatedQuery.resource_type.split(",")
+    selector.resource_type = values.length > 1 ? values : values[0]
+  }
+
+  if (validatedQuery.resource_id) {
+    const values = validatedQuery.resource_id.split(",")
+    selector.resource_id = values.length > 1 ? values : values[0]
+  }
+
+  if (validatedQuery.to) {
+    const values = validatedQuery.to.split(",")
+    selector.to = values.length > 1 ? values : values[0]
+  }
+
+  if (
+    !validatedQuery.include_resends ||
+    validatedQuery.include_resends === "false"
+  ) {
+    selector.parent_id = null
+  }
+
+  const listConfig = {
+    select: includeFields.length
+      ? includeFields
+      : defaultAdminNotificationsFields,
+    relations: expandFields.length
+      ? expandFields
+      : defaultAdminNotificationsRelations,
+    skip: offset,
+    take: limit,
+    order: { created_at: "DESC" },
+  }
+
+  const notifications = await notificationService.list(selector, listConfig)
+
+  const fields = [...listConfig.select, ...listConfig.relations]
+  const data = notifications.map((o) => _.pick(o, fields))
+
+  res.json({ notifications: data, offset, limit })
+}
+
+export class AdminGetNotificationsQuery {
```

Comment:
Maybe the `AdminGetNotificationsQuery` should be moved to types / `notifications` so that it can be reused in the service layer? Alike what is sort of done in the PR for migrating `orders` (specifically, the `selector` class exported [here](https://github.com/medusajs/medusa/blob/b53f73aa08e83bc155a961a80327669f661051c8/packages/medusa/src/types/orders.ts), and as proposed [here](https://github.com/medusajs/medusa/pull/790)).

Reply:
totally! however, thought it would be better for that to come a bit later on when we start migrating the service layer as well. What do you think?

Reply:
Might be! However, this is what we are currently doing in other PRs :-)